### PR TITLE
Fix deprecations in DiffRules >= 1.4

### PR DESCRIPTION
**Pull request by devmotion**
This PR fixes deprecations introduced by DiffRules 1.4, and in particular the unsafe use of `DiffRules.diffrules()` (currently no checks if the modules and functions exist at all and the definitions can be `@eval`ed).

I used the opportunity to load the rules for LogExpFunctions from DiffRules as well which were added in DiffRules 1.4. Recent versions of SpecialFunctions depend on LogExpFunctions and hence it is already an indirect dependency.

Basically an analogue of https://github.com/JuliaDiff/ReverseDiff.jl/pull/191.

### CODE REVIEW

**Comment by devmotion** [comment]
The deprecation warnings on the master branch (https://github.com/FluxML/Tracker.jl/runs/5013456661?check_suite_focus=true#step:6:38) are fixed by this PR: https://github.com/FluxML/Tracker.jl/runs/5014065869?check_suite_focus=true#step:6:37

**Comment by devmotion** [comment]
Bump :slightly_smiling_face: 

The missing DiffRules rules for LogExpFunctions cause some problems in Distributions/DistributionsAD, it would be nice to add them (and, of course, to fix the deprecation warnings).

**Review by ToucheSir — APPROVED** [review]
